pandas/tests/frame/methods/test_join.py

```diff
@@ -373,3 +373,11 @@ def test_frame_join_tzaware(self):
 
         tm.assert_index_equal(result.index, expected)
         assert result.index.tz.zone == "US/Central"
+
+
+def test_join_duplicate_indicies():
+    df1 = pd.DataFrame(np.random.randn(5), index=[0, 1, 2, 3, 3], columns=["a"])
```

Comment:
add `# GH xxxxx`, with the relevant issue your solving with the test above this line.